src/ng/parse.js

```diff
@@ -781,7 +819,7 @@ ASTCompiler.prototype = {
       filters: {},
       fn: {vars: [], body: [], own: {}},
       assign: {vars: [], body: [], own: {}},
-      inputs: []
+      inputs: {}
```

Comment:
By doing this (and later `Object.keys`) you are abusing the fact that `Object.keys` returns the keys in the order that were added to assert that expressions as `foo.bar();foo.baz()` will be evaluated in the right order. Even when this is somehow true, this is not part of the spec and we should not depend on this out-of-spec fact.